module/modmanager/manager.go

```diff
@@ -475,25 +464,34 @@ func (m *module) stopProcess() error {
 
 func (m *module) registerResources(mgr modmaninterface.ModuleManager, logger golog.Logger) {
 	for api, models := range m.handles {
-		known := registry.ResourceSubtypeLookup(api.Subtype)
-		if known == nil {
+		if _, ok := registry.ResourceSubtypeLookup(api.Subtype); !ok {
 			registry.RegisterResourceSubtype(api.Subtype, registry.ResourceSubtype{ReflectRPCServiceDesc: api.Desc})
 		}
 
 		switch api.Subtype.ResourceType {
 		case resource.ResourceTypeComponent:
 			for _, model := range models {
 				registry.RegisterComponent(api.Subtype, model, registry.Component{
-					Constructor: func(ctx context.Context, deps registry.Dependencies, cfg config.Component, logger golog.Logger) (interface{}, error) {
-						return mgr.AddResource(ctx, cfg, DepsToNames(deps))
+					Constructor: func(
+						ctx context.Context,
+						deps resource.Dependencies,
+						conf resource.Config,
+						logger golog.Logger,
+					) (resource.Resource, error) {
+						return mgr.AddResource(ctx, conf, DepsToNames(deps))
 					},
 				})
 			}
 		case resource.ResourceTypeService:
 			for _, model := range models {
-				registry.RegisterService(api.Subtype, model, registry.Service{
-					Constructor: func(ctx context.Context, deps registry.Dependencies, cfg config.Service, logger golog.Logger) (interface{}, error) {
-						return mgr.AddResource(ctx, config.ServiceConfigToShared(cfg), DepsToNames(deps))
+				registry.RegisterResource(api.Subtype, model, registry.Resource{
```

Comment:
Is there a reason why this is using RegisterResource directly (instead of the RegisterService helper) when the component case above is using the RegisterComponent helper?

Reply:
nope!